src/main/java/bio/terra/service/snapshot/SnapshotDao.java

```diff
@@ -421,14 +432,19 @@ Snapshot retrieveWorker(String sql, MapSqlParameterSource params) {
         // It seemed like the cleanest thing to me at the time.
         UUID projectResourceId = snapshot.getProjectResourceId();
         if (projectResourceId != null) {
-          snapshot.projectResource(
-              resourceService.getProjectResource(snapshot.getProjectResourceId()));
+          snapshot.projectResource(resourceService.getProjectResource(projectResourceId));
         }
 
         // Retrieve the Azure Storage Account associated with the snapshot.
         resourceService
             .getSnapshotStorageAccount(snapshot.getId())
             .ifPresent(snapshot::storageAccountResource);
+
+        // Retrieve the DUOS Firecloud group associated with the snapshot.
```

Comment:
Calling out that with the way this is written, we will always include the linked DUOS Firecloud group (if one exists) in the `retrieveSnapshot` response.

But another option is to expand the `SnapshotIncludeRetrieveModel` to allow the user to include / exclude it in the response.  LMK if this is preferred.